### PR TITLE
Upgrade primer/react

### PR DESCRIPTION
**Pull request by siddharthkp**
```diff
- "@primer/components": "^30.0.0",
+ "@primer/react": "^34.5.0",

- "eslint-plugin-primer-react": "0.5",
+ "eslint-plugin-primer-react": "0.7.3"
```

### CODE REVIEW

File: docs/package.json [hunk 1]
```diff
@@ -9,8 +9,8 @@
     "now-build": "npm run build"
   },
   "dependencies": {
-    "@primer/components": "^30.0.0",
     "@primer/gatsby-theme-doctocat": "*",
+    "@primer/react": "^34.2.0",
```
Comment (colebemis) [inline docs/package.json:13]:
```suggestion
    "@primer/react": "^34.5.0",
```

File: theme/package.json [hunk 2]
```diff
@@ -33,7 +33,7 @@
     "@mdx-js/mdx": "^1.0.21",
     "@mdx-js/react": "^1.0.21",
     "@primer/component-metadata": "^0.4.0",
-    "@primer/components": "^30.0.0",
+    "@primer/react": "^34.2.0",
```
Comment (colebemis) [inline theme/package.json:36]:
```suggestion
    "@primer/react": "^34.5.0",
```

**Review by pksjce — APPROVED** [review]
Only nit is that this seems to be mainly using components we've deprecated. Hopefully all the new components will work just as fine here. 👍

**Review by rezrah — APPROVED** [review]
🙌 ...

@pksjce - There's a bit of a circular dependency issue at play.

We need to move off @primer/components here in order to make the deprecations to those same components on @primer/react. I think this PR should now let us [remove the resolve alias for `@primer/components` in docs](https://github.com/primer/react/blob/1195336e475441f6f7bc266630070834ad564503/docs/gatsby-config.js#L22), which was presumably blocking your PR too.

**Comment by siddharthkp** [comment]
> Only nit is that this seems to be mainly using components we've deprecated

True that 😢

**Pull request merged**